### PR TITLE
Add log messages for mail errors

### PR DESCRIPTION
**Pull request by p-h-a-i-l**
## Proposed change

During fixing a tls issue (parallel work with https://github.com/paperless-ngx/paperless-ngx/pull/705) I noticed no errors were visible to the end user.
This fix adds error logging for mail errors.
Also a pinning of imap-tools is proposed to prevent future breaks.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by stumpylog** [comment]
This does look useful to narrow down errors.  But since #705 went into beta, this should probably wait until we merge beta back to dev, then get a rebase or merge.

**Comment by qcasey** [comment]
This doesn't appear to have any conflicts, is a rebase still necessary?

File: Pipfile
Comment (stumpylog) [inline Pipfile:22]: This is now a version regression compared to `dev` and will break testing there.  

I'm not sure I see a reason to lock this down anyway?  Our testing has improved to catch breaking changes in this area especially

**Review by stumpylog — CHANGES_REQUESTED** [review]

**Comment by stumpylog** [comment]
I'm not sure how there wasn't a conflict, I guess since the same lines changed in the same way?  Anyway, just one minor thing to me

File: Pipfile
Comment (p-h-a-i-l) [reply Pipfile:22]: But how can you react to breaking changes when these changes are pulled  automatically?
With tests you can detect them, but fixes are only upstream and all older versions break on installation

Comment (stumpylog) [reply Pipfile:22]: I'm sorry, I'm not following what you mean. 

Nothing is pulled in automatically, dependabot only informs us of new version releases.  And the test suite runs on those PRs to catch issues, besides us reading changelings.

And we've made the changes in dev for 0.54.0, so at least this should match that.

Comment (p-h-a-i-l) [reply Pipfile]: Maybe there is a misunderstanding on my side :)

I'll try to explain what I thought happens:

Pipfile is the leading definition from which all dependencies are installed. This file is part of every release and cannot be changed afterwards.
If a wildcard is defined, the latest package Version from the repository is pulled on installation.
Dependabot can monitor requirement files and create alerts on changes. If action is needed, it is taken for the next release.

Therefore old releases break if a wildcard is defined and a breaking change happens.

Comment (stumpylog) [reply Pipfile]: I see.  Thankfully, that won't happen, as the actual versions of dependencies are locked in a couple places `requirements.txt` and `Pipfile.lock`.

`requirements.txt` is used when building the Docker image.  So on a release, which is basically a Git tag named the right way, that tags version of requirements.txt will be baked into the image, and so it can't change.

`Pipfile.lock` is similar, as it is the source of requirements.txt, but includes more detailed information, as well as being useful for local development setup to create an isolated virtualenv.

All that is a long winded way to say the `Pipfile` is the list of dependencies while `requirements.txt` and `Pipfile.lock` lock down the actual versions installed.  So the only way an existing version could change library versions is through a user's manual intervention, and even that pretty much only in a bare metal install.

Thanks for setting this back to `0.54.0`

**Review by stumpylog — APPROVED** [review]

File: Pipfile
Comment (p-h-a-i-l) [reply Pipfile]: Thanks for explaining :) 
Looks like a lot of thought went into dependency management, I'll leave the decision to you whether to lock down imap-tools in Pipfile or not. Please let me know, I'll change this PR accordingly.

**Pull request merged**
